orchagent/qosorch.cpp

```diff
@@ -1,10 +1,798 @@
+#include "sai.h"
+
 #include "qosorch.h"
 #include "logger.h"
 
+#include <sstream>
+#include <iostream>
+
+extern sai_port_api_t      *sai_port_api;
+extern sai_queue_api_t     *sai_queue_api;
+extern sai_scheduler_api_t *sai_scheduler_api;
+extern sai_wred_api_t      *sai_wred_api;
+extern sai_qos_map_api_t   *sai_qos_map_api;
+
 namespace swss {
 
-void QosOrch::doTask(_in_ Consumer& consumer_info)
+
+qos_type_map QosOrch::m_qos_type_maps = {
+    {APP_DSCP_TO_TC_MAP_TABLE_NAME,  new qos_object_map()},
+    {APP_TC_TO_QUEUE_MAP_TABLE_NAME, new qos_object_map()},
+    {APP_SCHEDULER_TABLE_NAME,       new qos_object_map()},
+    {APP_WRED_PROFILE_TABLE_NAME,    new qos_object_map()}
+};
+
+
+bool QosMapHandler::processWorkItem(Consumer& consumer)
+{
+    sai_object_id_t         sai_object          = SAI_NULL_OBJECT_ID;
+    auto                    it                  = consumer.m_toSync.begin();
+    KeyOpFieldsValuesTuple  tuple               = it->second;
+    string                  qos_object_name     = kfvKey(tuple);
+    string                  qos_map_type_name   = consumer.m_consumer->getTableName();
+    string                  op                  = kfvOp(tuple);
+
+    if(!isValidTable(qos_map_type_name)) {
+        return false;
+    }
+
+    if(QosOrch::getTypeMap()[qos_map_type_name]->find(qos_object_name) != QosOrch::getTypeMap()[qos_map_type_name]->end()) {
+        sai_object = (*(QosOrch::getTypeMap()[qos_map_type_name]))[qos_object_name];
+    }
+    if(op == SET_COMMAND) {
+        std::vector<sai_attribute_t> attributes;
+        if(!convertFieldValuesToAttributes(tuple, attributes)) {
+            return false;
+        }
+        if(SAI_NULL_OBJECT_ID != sai_object) {
+            if(!modifyExistingSaiObject(sai_object, attributes)) {
+                SWSS_LOG_ERROR("Failed to set settings to existing dscp_to_tc map. db name:%s sai object:%llx", 
```

Comment:
Is this processWorkItem function a general function to all maps or just to dscp_to_tc map?